community/kernel/src/main/java/org/neo4j/kernel/impl/api/DataIntegrityValidatingStatementOperations.java

```diff
@@ -136,10 +136,13 @@ public void indexDrop( KernelStatement state, IndexDescriptor index ) throws Dro
 
             if ( existingIndex.type() == UNIQUE )
             {
-                throw new IndexBelongsToConstraintException( index.schema() );
+                if ( schemaReadDelegate.indexGetOwningUniquenessConstraintId( state, existingIndex ) != null )
+                {
+                    throw new IndexBelongsToConstraintException( index.schema() );
+                }
             }
         }
-        catch ( IndexBelongsToConstraintException | NoSuchIndexException e )
+        catch ( IndexBelongsToConstraintException | NoSuchIndexException | SchemaRuleNotFoundException e )
```

Comment:
That `SchemaRuleNotFoundException` looks strange, like the method might throw that instead of returning `null` when you look at just this part of the code. However, a quick glancing through the code indicates that this exception is never thrown from this level in the code.

Could we remove the exception from the `indexGetOwningUniquenessConstraintId` declaration and all related catch clauses, and update the javadocs to say that `null` is returned?